packages/web-components/fast-ssr/src/request-storage.ts

```diff
@@ -156,18 +184,30 @@ export const RequestStorageManager = Object.freeze({
 
     /**
      * Installs a DOM shim that ensures that window, document,
-     * and other globals are scoped per-request.
+     * and other globals are scoped per-request. Calling this function
+     * will have no effect if the shim has already been installed.
      */
     installDOMShim(): void {
         for (const key of perRequestGlobals) {
-            const original = (globalThis as any)[key];
-            Reflect.defineProperty(globalThis, key, {
-                get() {
-                    // Return original global variable if currently not in the storage scope
-                    const store = asyncLocalStorage.getStore() as Map<string, any>;
-                    return store ? store.get("window")[key] : original;
-                },
-            });
+            if (!shimIsInstalledFor(key)) {
+                preShimGlobals.set(key, (globalThis as any)[key]);
+                Object.defineProperty(globalThis, key, { get: perRequestGetters[key] });
+            }
+        }
+    },
+
+    /**
+     * Uninstalls the DOM shim installed by {@link RequestStorageManager.installDOMShim}.
+     * Calling this function will have no effect if there is no shim installed.
+     */
+    uninstallDOMShim(): void {
+        for (const key of perRequestGlobals) {
+            if (shimIsInstalledFor(key)) {
+                Object.defineProperty(globalThis, key, {
+                    value: preShimGlobals.get(key),
```

Comment:
Should the `value` be `get` instead since `get` was the one redefined in `installDOMShim`

Reply:
Hmm actually it should *probably* store the previous descriptor and not just the value so that it can be re-defined exactly as it was. I'll update that

Reply:
Fixed this up to re-define properties with the previous descriptor, otherwise it just deletes the property off the global during uninstallation